### ISSUE TITLE
What is more important "transparency", "reproducibility" or "replication"?

### ISSUE THREAD

**Issue by aaronpeikert** [aaronpeikert/repro-tutorial#108]
This comment of @cjvanlissa has provoked me to reread some sections of The Logic of Scientific Discovery (it was years ago that I read it more or less thoroughly:

> CJ More fundamental: Reproducibility is a cornerstone of the scientific METHOD. I think it's more important to emphasize this, rather than critical assessment of others' work. See Karl Popper, The Logic of Scientific Discovery, Routledge, London, 1992, p. 66. 

And to a lesser extent this one:

> CJ Not clear how transparency outweighs (?) concerns about inflated false positive rate.

And I must admit I come to opposite conclusions, that is "critical assessment of others' work" is viewed by Popper to be the cornerstone of the scientific method. I take the stance that transparency enables what Popper would call inter-subjective criticism and should be the chief priority behind all open science practices.
Indeed it lies at the centre of the claim to be objective and therefore at the heart of what it means to do empirical science.

I hope I have translated this conviction into #107 without overburdening the introduction. Anyway, I think this could be a fruitful discussion.

**Comment by cjvanlissa** [comment]
I disagree with the emphasis on critical assessment of others' work for several reasons:
1) It is bad advertising. "Join the open science movement and get criticized!" is a hard sell compared to "Join the open science movement and make your life easier, your work better, and meet cool new collaborators!" At least in the Netherlands, there is a lot of discomfort around the so-called "data police", people who make a name for themselves by bringing others' careers down. I don't want to raise the impression that we are encouraging this kind of policing, and would prefer to promote scientific progress by using the carrot rather than the stick.
2) Only a minority of reproducible work does end up being scrutinized, but ALL reproducible work accrues the benefits of being less likely to contain errors, easier to update, more trustworthy, easy to revisit after a few years, and easy to share with collaborators.
3) I find it hard to believe that Popper did/would not consider reproducibility (i.e., the research is not based on some intractable error in the workflow) to be more fundamental than critical assessment of the work of others. But if he didn't ... then I think he's wrong :p Basically the argument is: If your workflow is not reproducible, and all you have is some copy-pasted output from an analysis that "used to work", you have no credible evidence to support your claim - regardless of critical assessment by others.

**Comment by aaronpeikert** [comment]
We certainly agree on the first two points or the "practical side", which is certainly most important to convince researchers to work reproducibly. These alone are reasons enough to not emphasize transparency to much.
However, 3. is interesting from a more abstract standpoint.

My core argument is the following. Transparency is more important because something that is transparent is necessarily reproducible, but something that is reproducible can be intransparent e.g. a black box or binary program can produce the same numbers from the same data. All the positive aspects "like being less likely to contain errors, easier to update, more trustworthy, easy to revisit after a few years, and easy to share with collaborators" are not applicable to reproducible black boxes, but stem from transparent enough research. Reproducibility is just a tangible effect of being transparent enough.

Maybe it's just weird to say that something you created, can be transparent to yourself :)

Now, what would Popper say? 
To my knowledge, he never wrote about reproducibility in our definition (same data + same code + independent researcher = same results) so we will never know for sure.
My understanding is this:

He would consider an irreproducible statement (like you describe "all you have is some copy-pasted output from an analysis that "used to work") as **subjective** because it does not fit his definition of "objective" because only your experienced it and no one else can:

> I shall therefore say that the objectivity of scientific statements lies in the fact that they can be inter-subjectively tested.*1

In later editions he added a footnote about the meaning of  inter-subjectively tested:

> *1 I have since generalized this formulation; for inter-subjective testing is merely a very
important aspect of the more general idea of inter-subjective criticism, or in other words,
of the idea of mutual rational control by critical discussion. This more general idea,
discussed at some length in my Open Society and Its Enemies, chapters 23 and 24, and in my
Poverty of Historicism, section 32, is also discussed in my Postscript, especially in chapters *i,
*ii, and *vi.

So his harsh judgment of a not reproducible claim would be that it is **unscientific**.

>We may now return to a point made in the previous section: to my thesis that a subjective experience, or a feeling of conviction, can never justify a scientific statement, and that within science it can play no part [...]

So I dare say the argument for reproducibility is:

Empirical sciences -> must be objective -> must be transparent (my translation of "the theoretical possibility to be open to inter-subjective criticism") -> reproducibility

**Comment by cjvanlissa** [comment]
Maybe there's also a difference of experience here. You are a digital native, starting your PhD at a time when open science is taking hold, at an institution with exceptionally high standards for reproducibility. I am from a generation who just did their analyses "on the fly", never saving syntax, ending up after weeks/months of hacking with results that could not be reproduced, and then faced with a crushing decision: Whether to publish results we knew were not reproducible, or eat the lost time and start over. My first PhD paper took nearly 2 years to write, because after 1.5 years I discovered that the results I was writing about were not reproducible (still no idea what happened). I started over - but I think a lot of authors just push for publication even if the results are not reproducible. When asking authors for raw output in preparing meta-analyses, I got a lot of responses like "we don't have that output and we don't know how to recreate it". I think a lot of our potential readership will relate to this dilemma revolving around reproducibility, and that it's just the more fundamental/widespread issue when compared to critical evaluation of others' work.
